raiden/api/rest.py

```diff
@@ -358,16 +357,38 @@ def start(self, host='127.0.0.1', port=5001):
                 log=wsgi_log,
                 error_log=wsgi_log,
             )
-            self.wsgiserver.start()
+            self.wsgiserver.serve_forever()
         except socket.error as e:
             if e.errno == errno.EADDRINUSE:
                 raise APIServerPortInUseError()
             raise
+        finally:
+            self.wsgiserver.stop()
+            self.wsgiserver = None
+
+    def start(self, host='127.0.0.1', port=5001):
+        self.greenlet = gevent.spawn(self.run, host, port)
+        return self.greenlet
 
     def stop(self, timeout=5):
-        if getattr(self, 'wsgiserver', None):
-            self.wsgiserver.stop(timeout)
-            self.wsgiserver = None
+        if getattr(self, 'greenlet', None):
+            self.wsgiserver.stop(timeout=timeout)
```

Comment:
Why do we stop `wsgiserver` here? Isn't that supposed to happen in the `finally` block above after the `greenlet.get` in the next line?